### PR TITLE
DAOS-1449 object: daos_obj_fetch iod null causes segfault

### PR DESCRIPTION
**Pull request by GitHuaKuang**
Instructed daos_obj_fatch() call to inject NULL oid to test if it will cause segfault.

Randomly picked the call for IO13 for test purpose:
reading un-existed record ...
[ ERROR ] — Test failed with exception: Segmentation fault(11)

With the fix, got an error without segfault:
reading un-existed record ...
[  ERROR   ] --- 0xfffffffffffffc15 != 0
[   LINE   ] --- src/tests/suite/daos_obj.c:1943: error: Failure!

Change-Id: I48127f96d3fad953376e6d4ef78aa4742295529e
Signed-off-by: Hua Kuang <hua.kuang@intel.com>

### CODE REVIEW

**Comment by GitHuaKuang** [comment]
The failure is also occurred and  reported in:
https://jira.hpdd.intel.com/browse/DAOS-4410

@daos-stack/daos-gatekeeper Can we land this one line straight forward change?

File: src/object/cli_obj.c [hunk 1]
```diff
@@ -1419,6 +1419,9 @@ obj_iod_sgl_valid(unsigned int nr, daos_iod_t *iods, d_sg_list_t *sgls,
 	int	i;
 	int	rc;
 
+	if (iods == NULL)
+		return -DER_INVAL;
```
Comment (johannlombardi) [inline src/object/cli_obj.c:1423]: I think that the patch is fine, but i don't grok how this is related to the test that does not pass a NULL iod as far as i can see. Could you please elaborate?

Comment (GitHuaKuang) [reply src/object/cli_obj.c:1423]: Most our using test code do not pass a NULL, but this ticket is for code sanity test (not real running tests) to make sure the API is secure even if some malfunction call(hacker, stress test, etc) passing NULL:
DAOS-1449: "If I pass a null pointer for the IOD parameter it appears to segfault."

SO, this fix is to make sure even NULL oid is passed in, DAOS wouldn't crash.

**Comment by GitHuaKuang** [comment]
@mchaarawi I think the answer to Johann's comments should provide the same explanation to your question: The ticket was created to alert that this api is not handling this NULL parameter properly, even the existing callers are not passing in NULL oid, it's trying to make this API robust to any potential invalid input, future or current.

**Comment by mchaarawi** [comment]
> @mchaarawi I think the answer to Johann's comments should provide the same explanation to your question: The ticket was created to alert that this api is not handling this NULL parameter properly, even the existing callers are not passing in NULL oid, it's trying to make this API robust to any potential invalid input, future or current.

Your comment is referring to IO-13 error which does not pass a NULL iods. so i will say again, that the fix is correct but the comment is wrong.
What you should do is add a **new** test in daos_obj.c that passes a NULL and exercise this fix.

**Comment by GitHuaKuang** [comment]
@mchaarawi I updated the comment to clarify the use of IO13, it's just used as test purpose, hope now the comment has avoided implying IO13 is having this issue.

**Pull request merged**